test/models/address_test.rb

```diff
@@ -3,7 +3,17 @@
 require 'test_helper'
 
 class AddressTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test 'normalizes postcode' do
+    Geocoder.stub :search, [] do
+
+      address = Address.create(
+        street_address: '123 street',
+        postcode: '   ha8 5ha ',
```

Comment:
maybe add a few more test cases here including postcodes like "W1W 5AA" and "NG12 4FZ" with various spacing and also spacing between the two parts of it